extensions/parquet/compression/src/main/java/io/deephaven/parquet/compress/DeephavenCompressorAdapterFactory.java

```diff
@@ -183,21 +237,30 @@ private DeephavenCompressorAdapterFactory(CompressionCodecFactory compressionCod
      * @param codecName the name of the codec to search for.
      * @return a compressor instance with a name matching the given codec.
      */
-    public CompressorAdapter getByName(String codecName) {
+    public CompressorAdapter getByName(final String codecName) {
         if (codecName.equalsIgnoreCase("UNCOMPRESSED")) {
             return CompressorAdapter.PASSTHRU;
         }
-        final CompressionCodec codec = compressionCodecFactory.getCodecByName(codecName);
+        CompressionCodec codec = compressionCodecFactory.getCodecByName(codecName);
         if (codec == null) {
-            throw new IllegalArgumentException(
-                    String.format("Failed to find CompressionCodec for codecName=%s", codecName));
+            if (codecName.equalsIgnoreCase("LZ4_RAW")) {
+                // Hacky work-around since codec factory refers to LZ4_RAW as LZ4RAW
+                codec = compressionCodecFactory.getCodecByName("LZ4RAW");
+            }
+            if (codec == null) {
+                throw new IllegalArgumentException(
+                        String.format("Failed to find CompressionCodec for codecName=%s", codecName));
+            }
         }
         final CompressionCodecName ccn = codecClassnameToCodecName.get(codec.getClass().getName());
         if (ccn == null) {
             throw new IllegalArgumentException(String.format(
                     "Failed to find CompressionCodecName for codecName=%s, codec=%s, codec.getDefaultExtension()=%s",
                     codecName, codec, codec.getDefaultExtension()));
         }
+        if (ccn == CompressionCodecName.LZ4) {
```

Comment:
can't we use the configuration in createInstance() to replace this?

Reply:
Oh I see - this is specifically to say "take the expected LZ4, but wrap it in the backup impl (which will in turn look up a lz4raw instance from config if needed". Otherwise, we have to hardcode the lz4 impl (since we don't want it to look itself up).

Reply:
Yup, exactly.